happi/loader.py

```diff
@@ -72,10 +74,22 @@ class is then instantiated with the given args and kwargs provided.
     attach_md: bool, optional
         Attach the container to the instantiated object as `md`
 
+    use_cache: bool, optional
+        When devices are loaded they are stored in the ``happi.cache``
+        dictionary. This means that repeated attempts to load the device will
+        return the same object. This prevents unnecessary EPICS connections
+        from being initialized in the same process. If a new object is
+        needed, set `use_cache` to False and a new object will be created,
+        overriding the current cached object
+
     Returns
     -------
     obj : happi.Device.device_class
     """
+    # Return a cached version of the device if present and not forced
+    if use_cache and device.prefix in cache:
+        return cache[device.prefix]
```

Comment:
Devil's advocate for a second: will two devices ever have the same prefixes? I can't immediately think of any cases, but it's good to worry. If this is a fundamental assumption of `happi` that's totally good (is it the `_id` field?), I just want to make sure.

Reply:
Yeah. `happi` assumes that you won't ever have two devices with the same prefix. I thought about making it a hash of name and prefix, but decided we shouldn't allow two devices with the same prefix.